lib/application-stack.ts

```diff
@@ -299,17 +299,68 @@ export class Oncoanalyser extends Construct {
       `${props.bucket.outputPrefix}/*`,
     );
 
-    const nextflowConfigTemplate = fs.readFileSync(path.join(__dirname, "resources/nextflow_aws.template.config"), { encoding: "utf-8"});
+    const createProcessImageAsset = (env: any, envName: string, cdkId: string, imageName: string) => {
+      const imageAsset = new ecrAssets.DockerImageAsset(this, cdkId, {
+        directory: path.join(__dirname, 'process_docker_images'),
+        platform: Platform.LINUX_AMD64,
+        // because the image base name is passed into Docker - the actual Docker checksum
+        // itself won't change even when the image base does... so we need to add it into the hash
+        extraHash: imageName,
+        buildArgs: {
+          // pass this through to Docker forming the base of the image we are constructing
+          BASE_IMAGE: imageName,
+        }
+      });
 
-    const nextflowConfigTemplateCompiled = Handlebars.compile(nextflowConfigTemplate);
+      env[envName] = imageAsset.imageUri;
+    }
 
-    const nextflowConfig = nextflowConfigTemplateCompiled({
+    const env: Record<string,string> = {
       BATCH_INSTANCE_TASK_ROLE_ARN: roleBatchInstanceTask.roleArn,
       BATCH_JOB_QUEUE_NAME: jobQueueTask.jobQueueName,
       S3_BUCKET_NAME: props.bucket.bucket,
       S3_BUCKET_REFDATA_PREFIX: props.bucket.refDataPrefix,
-      BATCH_VOLUME_MOUNT_POINT: BATCH_VOLUME_MOUNT_POINT
-    }, { });
+      BATCH_VOLUME_MOUNT_POINT: BATCH_VOLUME_MOUNT_POINT,
+    };
+
+      // modules/local/neo/annotate_fusions/main.nf:                   'biocontainers/hmftools-isofox:1.7.1--hdfd78af_1'
+    // modules/local/neo/scorer/main.nf:                             'biocontainers/hmftools-neo:1.2--hdfd78af_1'
+    // modules/local/neo/finder/main.nf:                             'biocontainers/hmftools-neo:1.2--hdfd78af_1'
+    // modules/local/bamtools/main.nf:                               'biocontainers/hmftools-bam-tools:1.3--hdfd78af_0'
+    // modules/local/linxreport/main.nf:                             'biocontainers/r-linxreport:1.0.0--r43hdfd78af_0'
+    // modules/local/pave/germline/main.nf:                          'biocontainers/hmftools-pave:1.7--hdfd78af_0'
+    // modules/local/pave/somatic/main.nf:                           'biocontainers/hmftools-pave:1.7--hdfd78af_0'
+
+    createProcessImageAsset(env, "ESVEE_DOCKER_IMAGE_URI", "EsveeImageAsset", "quay.io/biocontainers/hmftools-esvee:1.0--hdfd78af_0")
```

Comment:
Is there any kind of dependency/manifest file that the versions and package names could be read from? I feel like it might tedious to update this when the versions change.

Reply:
Otherwise it might be possible to run the dockerfile in a `dry_run` mode, see what it needs, and then use that to fetch the other docker images.

This might be more complex than we want though.